### PR TITLE
Added feeds related to the whoishiring bot

### PR DESCRIPTION
**Pull request by jaredandrews**
Every month the HN bot [whoishiring](https://news.ycombinator.com/submitted?id=whoishiring) posts two threads that I am interested in:

* Ask HN: Who is hiring? ([most recent here](https://news.ycombinator.com/item?id=15824597))
* Ask HN: Freelancer? Seeking freelancer? ([most recent here](https://news.ycombinator.com/item?id=15824598))

**Use case**
I think the responses to these threads are worthy of their own HNRSS feeds. A job hunter or freelancer with an RSS feedreader can hook it up and search for relevant keywords.

**What is in this Pull Request**
Right now HNRSS can provide a feed of the threads started by the `whoishiring` user. It can also provide a feed of responses to those threads. This Pull Request automates the middle step here which is getting the id of the threads started by `whoishiring` and feeding them to the `comments` endpoint that already exists. Multiple posts by `whoishiring` are merged together to create a feed of responses across all posts.

I have added three endpoints:
* `/whoishiring` - returns an aggregate feed of all thread responses by `whoishiring`
* `/whoishiring/jobs` - returns an aggregate feed of all thread responses to 'Ask HN: Who is hiring?' by `whoishiring`
* `/whoishiring/freelance` - returns an aggregate feed of all thread responses to 'Ask HN: Freelancer? Seeking freelancer? ' by `whoishiring`

Please let me know what you think. Thanks!

### CODE REVIEW

File: api.py [hunk 1]
```diff
@@ -96,3 +96,36 @@ def user(self, username, include='all'):
...
+            api = API()
+            api_response = api.comments(story_id=thread_id)
+            if 'hits' in api_response:
+                all_hits.extend(api_response.get('hits'))
+
+            all_responses = merge_two_dicts(api_response, all_responses)
+
+        pseudo_response = {}
+        pseudo_response[u'hits'] = all_hits
+        pseudo_response['link_to'] = None
```
Comment (jaredandrews) [inline api.py]: This is a little sketchy, open to ideas for a 'cleaner' approach.

File: api.py [hunk 2]
```diff
@@ -96,3 +96,36 @@ def user(self, username, include='all'):
...
+
+    def who_is_hiring(self, include='all'):
+        submitted = self.user('whoishiring', 'submitted')
+
+        thread_ids = []
+        for item in submitted.get('hits'):
+            if 'Ask HN: Who is hiring?' in item[u'title'] and ('all' in include or 'jobs' in include):
+                thread_ids.append(item.get('objectID'))
+            elif 'Ask HN: Freelancer? Seeking freelancer?' in item[u'title'] and ('all' in include or 'freelance' in include):
+                thread_ids.append(item.get('objectID'))
```
Comment (jaredandrews) [inline api.py]: This basically gets **all** the threads started by `whoishiring` and filters them.

File: api.py [hunk 3]
```diff
@@ -96,3 +96,36 @@ def user(self, username, include='all'):
...
+
+        all_hits = []
+        all_responses = {}
+        for thread_id in thread_ids:
+            api = API()
+            api_response = api.comments(story_id=thread_id)
+            if 'hits' in api_response:
+                all_hits.extend(api_response.get('hits'))
+
+            all_responses = merge_two_dicts(api_response, all_responses)
```
Comment (jaredandrews) [inline api.py]: The threads are merged into a "pseudo response" that can be read by `rss.py` and turned into an RSS feed.

File: api.py [hunk 4]
```diff
@@ -96,3 +96,36 @@ def user(self, username, include='all'):
...
+
+        pseudo_response = {}
+        pseudo_response[u'hits'] = all_hits
+        pseudo_response['link_to'] = None
+
+        return pseudo_response
```
Comment (jaredandrews) [inline api.py]: With a clean cache this entire process can take a couple seconds. I'm not sure if this is a problem on your end or not. Speed could be increased by limiting the number of threads from `whoishiring` which are analyzed.

Comment (jaredandrews) [reply api.py]: The way GitHub shows comments makes this hard to parse. I am referring to the complete execution of `who_is_hiring` above.

**Comment by edavis** [comment]
Wow, this is a really good PR. The code looks clean, the documentation/explanations are top-notch, and this looks like an interesting and useful addition. I definitely see now you were going for something different than what I originally thought... but this looks great.

Give me a few days so I can find some time to dig into this more fully and work on getting it merged into master.

Thanks!

**Comment by jaredandrews** [comment]
@edavis thank you for the kind words! Let me know if you have any questions on the PR or want me to change any thing.

**Comment by jaredandrews** [comment]
@edavis hope you are having a wonderful 2018. Have you had a chance to check this out? Thanks!

**Comment by edavis** [comment]
@jaredandrews Yikes, I totally dropped the ball here!

I haven't forgotten about this. I planned on merging it in during the work lull between Christmas and New Years but my dumb body decided to get sick during that week instead.

Hopefully I can get to this soon. Worst case should be this weekend but hoping to knock it out in next few days.

Thanks for your patience!

**Comment by edavis** [comment]
Hey @jaredandrews 

This is up on hnrss.org now. I ended up making a few tweaks (bf44065dd5b3d17962467f0c2e36f96770e8929a for details), but this wouldn't have happened without your PR. So thanks for taking the time to send me that.

Kick the tires a bit and let me know if this works for you. Here are the URLs:

https://hnrss.org/whoishiring
https://hnrss.org/whoishiring/jobs
https://hnrss.org/whoishiring/freelance
https://hnrss.org/whoishiring/hired

If all looks good, I'll update the docs in a few days announcing it (with credit, of course).

Have a good one!

**Comment by jaredandrews** [comment]
@edavis Thanks man! Everything appears to be working. Will let you know if I notice any issues. Take it easy!
